From 14288802b8156d4ad9f24705469564aabe2b0997 Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Mon, 23 Sep 2024 09:49:59 +0200
Subject: [PATCH] CMake: Fix build on Solaris the stack_protector feature
 enabled

Solaris requires the ssp library to be linked when
-fstack-protector-strong is passed to the compiler.

Pick-to: 6.8
Fixes: QTBUG-129085
Change-Id: I55396fd9dc102ffd98bef54fc3dfc1aadf18a404
---
 cmake/QtInternalTargets.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/QtInternalTargets.cmake b/cmake/QtInternalTargets.cmake
index 5e1ea19ea0d8..bcfa8124596e 100644
--- qt-everywhere-src-6.8.0/qtbase/cmake/QtInternalTargets.cmake
+++ qt-everywhere-src-6.8.0/qtbase/cmake/QtInternalTargets.cmake
@@ -333,6 +333,9 @@ endif()
 
 if(QT_FEATURE_stack_protector)
     target_compile_options(PlatformCommonInternal INTERFACE -fstack-protector-strong)
+    if(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+        target_link_libraries(PlatformCommonInternal INTERFACE ssp)
+    endif()
 endif()
 
 if(QT_FEATURE_stack_clash_protection)
-- 
2.16.3

